src/hooks.ts

```diff
@@ -0,0 +1,25 @@
+import { getInitialLocale } from '$lib/utils/i18n/initialization';
+import { addLocaleToResponseBody, addLocaleRelativeRoutePath } from '$lib/utils/i18n/modifier';
+
+/** @type {import('@sveltejs/kit').GetContext} */
+export async function getContext(request) {
+	return {
+		locale: getInitialLocale(request)
+	};
+}
+
+/** @type {import('@sveltejs/kit').GetSession} */
+export function getSession({ context }) {
+	return {
+		locale: context.locale
+	};
+}
+
+/** @type {import('@sveltejs/kit').Handle} */
+export async function handle(request, render) {
+	addLocaleRelativeRoutePath(request);
```

Comment:
Using [`handle`](https://kit.svelte.dev/docs#hooks-handle) we can transform `request.path` before svelte-kit tries to find the route in the filesystem. Here, using [`addLocaleRelativeRoutePath`](https://github.com/kobejean/svelte-kit-i18n/blob/e32df541b2388d740a8109f4a7270129f3c8ef36/src/lib/utils/i18n/modifier.ts#L9-L14):
https://github.com/kobejean/svelte-kit-i18n/blob/e32df541b2388d740a8109f4a7270129f3c8ef36/src/lib/utils/i18n/modifier.ts#L9-L14
 I'm transforming paths like:
-  `/en` -> `/`
-  `/en/about` -> `/about`
-  `/ja/events/event` -> `/event`
so that way we can have the locale in the path but in the filesystem, files and directories are laid out how you normally would without a locale param in the route. 

You could probably do some more fancy mapping like:
-  `/en/about` -> `/about`
-  `/ja/情報` -> `/about`

Reply:
The one thing to be careful about is that when we do this mapping svelte-kit has no idea what the original path was and so when you look at `page.path` from `import { page } from '$app/stores';` it won't include the locale segment of the path.